hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientAbstract.java

```diff
@@ -1105,45 +1107,54 @@ public void testPutKey() throws IOException {
   void rewriteKey(BucketLayout layout) throws IOException {
     OzoneBucket bucket = createBucket(layout);
     OzoneKeyDetails keyDetails = createTestKey(bucket);
+    OmKeyArgs keyArgs = toOmKeyArgs(keyDetails);
+    OmKeyInfo keyInfo = ozoneManager.lookupKey(keyArgs);
 
     final byte[] newContent = "rewrite value".getBytes(UTF_8);
     rewriteKey(bucket, keyDetails, newContent);
 
     OzoneKeyDetails actualKeyDetails = assertKeyContent(bucket, keyDetails.getName(), newContent);
     assertThat(actualKeyDetails.getGeneration()).isGreaterThan(keyDetails.getGeneration());
-    assertEquals(keyDetails.getOwner(), actualKeyDetails.getOwner());
+    assertMetadataUnchanged(keyDetails, actualKeyDetails);
+    assertMetadataAfterRewrite(keyInfo, ozoneManager.lookupKey(keyArgs));
   }
 
   @ParameterizedTest
   @EnumSource
   void overwriteAfterRewrite(BucketLayout layout) throws IOException {
     OzoneBucket bucket = createBucket(layout);
     OzoneKeyDetails keyDetails = createTestKey(bucket);
+    OmKeyArgs keyArgs = toOmKeyArgs(keyDetails);
```

Comment:
NIT: I think we can remove this line since it is not being used.
```suggestion
```

Reply:
Thanks, removed leftover variable.